e2e-tests/ukef/cypress/e2e/journeys/utilisation-reports/utilisation-report-submission.spec.js

```diff
@@ -0,0 +1,235 @@
+    cy.wrap(visibleBanks).each((bank) => {
+      const reportPeriod = getPreviousReportPeriodForBankScheduleByMonth(bank.utilisationReportPeriodSchedule, submissionMonthStamp);
+      const mockUtilisationReport = UtilisationReportEntityMockBuilder.forStatus(UTILISATION_REPORT_RECONCILIATION_STATUS.REPORT_NOT_RECEIVED)
+        .withId(bank.id)
+        .withBankId(bank.id)
+        .withReportPeriod(reportPeriod)
+        .build();
+      cy.task(NODE_TASKS.INSERT_UTILISATION_REPORTS_INTO_DB, [mockUtilisationReport]);
+    });
+
+    // Insert a NOT_RECEIVED report for February 2023 for the bank the portal user belongs to
+    const notReceivedFebruary2023Report = UtilisationReportEntityMockBuilder.forStatus(UTILISATION_REPORT_RECONCILIATION_STATUS.REPORT_NOT_RECEIVED)
+      .withId(BANK1_PAYMENT_REPORT_OFFICER1.bank.id + 10000000)
+      .withBankId(BANK1_PAYMENT_REPORT_OFFICER1.bank.id)
+      .withReportPeriod({ start: { month: 2, year: 2023 }, end: { month: 2, year: 2023 } })
+      .build();
+    cy.task(NODE_TASKS.INSERT_UTILISATION_REPORTS_INTO_DB, [notReceivedFebruary2023Report]);
+
+    cy.clearCookie('dtfs-session');
+    cy.clearCookie('_csrf');
+    cy.getCookies().should('be.empty');
+  });
+
+  after(() => {
+    cy.clearCookies();
+    cy.clearCookie('dtfs-session');
+    cy.clearCookie('_csrf');
+    cy.getCookies().should('be.empty');
+
+    cy.task(NODE_TASKS.DELETE_ALL_FROM_SQL_DB);
+    cy.task(NODE_TASKS.DELETE_ALL_TFM_FACILITIES_FROM_DB);
+  });
+
+  /**
+   * Asserts the values in a given row of the premium payments table match the expected
+   * values from the report as defined in february2023ExpectedValues
+   * @param {number} premiumPaymentsRowIndex - The index of the row of the premium payments table
+   * @param {object} expectedParsedValuesForReportRow - The expected values to be found in the row
+   */
+  const assertPremiumPaymentsTableRowContainsExpectedValues = (premiumPaymentsRowIndex, expectedParsedValuesForReportRow) => {
+    cy.assertText(
+      tfmPages.utilisationReportPage.premiumPaymentsTab.premiumPaymentsTable.facilityIdByRowIndex(premiumPaymentsRowIndex),
+      expectedParsedValuesForReportRow.facilityId,
+    );
+    cy.assertText(
+      tfmPages.utilisationReportPage.premiumPaymentsTab.premiumPaymentsTable.exporterIdByRowIndex(premiumPaymentsRowIndex),
+      expectedParsedValuesForReportRow.exporter,
+    );
+    cy.assertText(
+      tfmPages.utilisationReportPage.premiumPaymentsTab.premiumPaymentsTable.reportedFeesByRowIndex(premiumPaymentsRowIndex),
+      expectedParsedValuesForReportRow.reportedFees,
+    );
+    cy.assertText(
+      tfmPages.utilisationReportPage.premiumPaymentsTab.premiumPaymentsTable.reportedPaymentsByRowIndex(premiumPaymentsRowIndex),
+      expectedParsedValuesForReportRow.reportedPayments,
+    );
+    cy.assertText(
+      tfmPages.utilisationReportPage.premiumPaymentsTab.premiumPaymentsTable.totalReportedPaymentsByRowIndex(premiumPaymentsRowIndex),
```

Comment:
Can we destructure the selectors (whilst keeping some level of context) so that it reads nicer? e.g

```js
cy.assertText(rowIndex.reportedFees(rowIndex), expectedValues.reportedFees);
```

Reply:
I didn't destructure because we have pages for both portal and tfm in this file and I felt it was quite useful to see at a glance which route we are currently in throughout the test. I think this is the pattern in the other 'ukef' e2e-tests